### DISCUSSION TITLE
Should Postgres subquery errors return 400s?

### DISCUSSION THREAD

**Discussion by anngbaum** [supabase/supabase#20964] [Questions] [answered]
I've had a few issues lately where Postgres RPCs encounter the error:
`more than one row returned by a subquery used as an expression`

Those bad requests returned `400` error codes to the clients.

Is that `400` to be expected? It was unintuitive to me because the error showed up as an `ERROR` in `postgres-logs` but a 400 in `edge-logs`. I usually filter `edge-logs` to only look at `500`s when debugging because we sometimes have old clients get out of whack for auth, and that muddies the `400` logs.

Given that the query's error was due to bad logic Postgres, I think I would have preferred/expected a `500` – just wanted to confirm that's expected behavior.

**Comment by GaryAustin1** [comment]
The REST API is PostgREST.  Here is the chart of when it returns what error code.  https://postgrest.org/en/stable/references/errors.html#http-status-codes

**Reply by anngbaum** [reply]
that's super helpful – I hadn't seen that mapping before. it seems like there are a few more reasons it raises a `400` than I had realized, specifically `foreign_key_violation` and `unique_violation` and `read_only_sql_transaction` all seem like things to monitor. thank you!

**Comment by steve-chavez** [comment, marked as answer]
> Given that the query's error was due to bad logic Postgres, I think I would have preferred/expected a 500 

Right, thanks for reporting. I've opened an issue here https://github.com/PostgREST/postgrest/issues/3205. A fix is already in progress https://github.com/PostgREST/postgrest/pull/3206.

**Reply by anngbaum** [reply]
that's awesome, thanks so much for the speedy response!
